homeassistant/components/geniushub/config_flow.py

```diff
@@ -0,0 +1,173 @@
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+
+        except socket.gaierror:
+            errors["base"] = "invalid_host"
+
+        except aiohttp.ClientResponseError as err:
+            if err.status == HTTPStatus.UNAUTHORIZED:
+                errors["base"] = "unauthorized"
+            else:
+                errors["base"] = "invalid_host"
+
+        except TimeoutError:
+            errors["base"] = "cannot_connect"
+
+        except aiohttp.ClientConnectionError:
+            errors["base"] = "cannot_connect"
+
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            _LOGGER.debug(
+                "ConfigFlow.async_step_option_1: validation passed:",
+                extra=user_input,
+            )
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        _LOGGER.debug(
+            "ConfigFlow.async_step_option_1: validation failed: ",
+            extra=user_input,
+        )
+
+        return self.async_show_form(
+            step_id="user",
+            errors=errors,
+        )
+
+    async def async_step_option_2(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Version 1 configuration."""
+        _LOGGER.debug("ConfigFlow.async_step_option_2: ", extra=user_input)
+        if user_input is None:
+            return self.async_show_form(step_id="option_2", data_schema=V1_API_SCHEMA)
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+
+        except aiohttp.ClientResponseError as err:
+            if err.status == HTTPStatus.UNAUTHORIZED:
+                errors["base"] = "unauthorized_token"
+            else:
+                errors["base"] = "invalid_host"
+
+        except socket.gaierror:
+            errors["base"] = "invalid_host"
+
+        except TimeoutError:
+            errors["base"] = "cannot_connect"
+
+        except aiohttp.ClientConnectionError:
+            errors["base"] = "cannot_connect"
+
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            _LOGGER.debug(
+                "ConfigFlow.async_step_option_2: validation passed: ",
+                extra=user_input,
+            )
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        _LOGGER.debug(
+            "ConfigFlow.async_step_option_2: validation failed: ",
+            extra=user_input,
+        )
+
+        return self.async_show_form(
+            step_id="user",
+            errors=errors,
+        )
+
+    async def async_step_import(self, user_input: dict[str, Any]) -> ConfigFlowResult:
```

Comment:
Also makes me wonder, if I understood correctly we used to have Mac adresses. Do we still need them? Can we use them as unique id for the config entry?

Reply:
@manzanotti can you comment about the Mac address? 

In the documentation:
_If required, you can switch between one Option and the other and, as the unique_id remains consistent, state history will be preserved. This assumes that the correct MAC address is provided for Option 2, below. If a wrong MAC address was provided for Option 1, then the MAC address can be overridden for Option 1 to maintain these links within the entity registry._

I've never needed to use the MAC address.

Happy to keep it in. Removing cleans up the code a little.

Reply:
At a guess, storing the MAC would make it easier for users with more than one hub, wouldn't it? I don't know if we actually support multiple hubs in this integration yet, but the official app does,.

Reply:
Each hub has to have the following:
1. A unique IP address
2. A unique MAC address
3. A unique token
At one time I am guessing a MAC address was required. The MAC address can be obtained from V1 or V3 API. Asking the user for it could be deprecated. 

The MAC address is optional for V3 and required for V1 (although the documentation states that it is optional).

I am going to leave it in for now as it does no harm and is getting beyond the scope of this PR.

Reply:
Checkout the comment in the PR discussion where I propose something and also talk about the scope